src/sage/numerical/backends/matrix_backend.pyx

```diff
@@ -0,0 +1,1170 @@
+r"""
+Matrix Backend
+
+It stores the problem data in Sage matrices. It allows users to specify a base ring.
+
+The class does not provide a solver method. It can be used as a base class for
+other classes implementing solvers.
+
+"""
+
+#*****************************************************************************
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#                  http://www.gnu.org/licenses/
+#*****************************************************************************
+
+from sage.matrix.constructor import Matrix as matrix
+from .generic_backend cimport GenericBackend
+import numpy
+
+cdef class MatrixBackend(GenericBackend):
+    """
+    Cython constructor
+
+    There is no support for integer variables.
+
+    EXAMPLES::
+
+        sage: p = MixedIntegerLinearProgram(solver="Matrix")
+
+    TESTS:
+
+    :trac:`20332`::
+
+        sage: p                                                     
+        Mixed Integer Program (no objective, 0 variables, 0 constraints)
+    """
+    def __cinit__(self, maximization = True, base_ring=None, numpy_implementation = False, **kwds):
+        """
```

Comment:
let's please get rid of the keyword `numpy_implementation` and instead accept a keyword `implementation`, which can be set to `"numpy"`.

Also add a doctest for this.